prow/crier/reporters/github/reporter.go

```diff
@@ -159,11 +164,19 @@ func (c *Client) Report(ctx context.Context, log *logrus.Entry, pj *v1.ProwJob)
 		if err := lock.Acquire(ctx, 1); err != nil {
 			return nil, nil, err
 		}
+
+		prRefGetter := config.NewRefGetterForGitHubPullRequest(c.ghc, pj.Spec.Refs.Org, pj.Spec.Refs.Repo, pj.Spec.Refs.Pulls[0].Number)
+		presubmits, err := c.config().GetPresubmits(c.gitClient, pj.Spec.Refs.Org+"/"+pj.Spec.Refs.Repo, prRefGetter.BaseSHA, prRefGetter.HeadSHA)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to get Presubmits for pull request %s/%s#%d: %v", pj.Spec.Refs.Org, pj.Spec.Refs.Repo, pj.Spec.Refs.Pulls[0].Number, err)
+		}
+		presubmit = &presubmits[0]
```

Comment:
I don't understand this, how does the `GetPresubmits` get the one exact presubmit config that we need (the one matching the prowjob we report?

Reply:
Fixed.